assets/js/common-components/relatedProducts.js

```diff
@@ -0,0 +1,53 @@
+var storefrontApp = angular.module('storefrontApp');
+
+storefrontApp.component('vcRelatedProducts', {
+    templateUrl: "themes/assets/js/common-components/relatedProducts.tpl.html",
+    bindings: {
+        productIds: '<',
+        responsive: '<',
+        onUpdate: '&'
+    },
+    controller: ['$timeout', '$scope', '$element', 'loadingIndicatorService', 'recommendationService', function ($timeout, $scope, $element, loader, recommendationService) {
+        var $ctrl = this;
+        $ctrl.loader = loader;
+
+        $ctrl.initCarousel = function () {
+            var responsive = Object.assign($ctrl.responsive);
+            Object.keys(responsive).map((key) => responsive[key].loop = responsive[key].items < $ctrl.products.length);
+            $timeout(function () {
+                $scope.$carousel = $element.find(".owl-carousel");
+                $scope.$carousel.on('initialized.owl.carousel refreshed.owl.carousel', function (event) {
+                    $scope.data = event;
+                    $ctrl.onUpdate({ $event: event });
+                });
+                $scope.$carousel.owlCarousel({
+                    nav: false,
+                    dots: false,
+                    responsive: responsive
+                });
+                // Temporary workaround for fallback-src
```

Comment:
Do we need this comment here? If it related to $timeout, maybe better to move it before $timeout defining?

Reply:
Owl carousel will copy first and last item for infinity loop.
It will copy ready html code and angular will not work on it.
For some reason, fallback-src works AFTER timout, while it should before.